packages/react-aria-components/test/ComboBox.test.js

```diff
@@ -34,15 +34,17 @@ let TestComboBox = (props) => (
 
 describe('ComboBox', () => {
   it('provides slots', () => {
-    let {getByRole} = render(<TestComboBox />);
+    let {getByRole, getAllByTestId} = render(<TestComboBox />);
 
     let input = getByRole('combobox');
     expect(input).toHaveValue('C');
 
     let combobox = input.closest('.react-aria-ComboBox');
-    expect(combobox).toHaveAttribute('data-foo', 'bar');
+    expect(combobox).toBeTruthy();
 
     expect(input).toHaveAttribute('aria-labelledby');
+    expect(input).toHaveAttribute('data-testid', 'bar');
```

Comment:
Note this change here. Previously getRenderProps was allowing through the data-* attributes through and were spreading them on the top level div, but many of the component's hooks actually handle propagating DOMProps to the inner input/group/etc. This resulted in duplicate data-* attributes on the outer div and the inner elements. I've removed the filterDOMProps spread on the outer div where applicable so this falls in line with how the Spectrum components behave, up for discussion if that is what we actually want or not.

Reply:
maybe good to assert `expect(input).toBe(getByTestId('bar'))`
also, if you use `getByTestId`, it'll throw and fail if it gets more than one result, so you don't need the expect for the length

Reply:
ok, after looking at the other example in the file below, this is more odd than I was thinking. I think since people explicitly provide the Input, we shouldn't handle it like RSP and move the data-testid from the Combobox down to the input. There should only be one copy though, so my comment about using getByTestId instead still stands

would like more opinions though

Reply:
Yeah, I'm a bit torn because the outer div isn't very important. For the ComboBox case currently, a user may pass   an `aria-label` or a `aria-labelledby` to the ComboBox like so:
```
  <ComboBox aria-label="foo" aria-labelledby="bar" data-testid="baz">
    <Label style={{display: 'block'}}>Test</Label>
    <div style={{display: 'flex'}}>
      <Input />
      <Button>
        <span aria-hidden="true" style={{padding: '0 2px'}}>▼</span>
      </Button>
    </div>
...
```
in which it is correctly propagated to the inner Label, all with the proper id concatenation due to useComboBox. However, would we expect those value to be applied to the outer div instead and shift the expectation that if the user wanted to apply their own aria attributes to the Input that they would do so directly? I feel like that might be something overlooked/not well understood by the user.

Reply:
Talked with team, will handle this in a separate PR. We'll want to filter out the `data-*` attributes from the props provided to the hooks so that they don't get duplicated onto things like inputProps. Will open a ticket for that